codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -420,9 +489,35 @@ public boolean visit(Http2Stream child) throws Http2Exception {
     }
 
     /**
+     * A special {@link DefaultState} designed to be used for the connection stream.
+     */
+    private final class DefaultConnectionState extends DefaultState {
```

Comment:
I don't think we need this class.  I suspect the code would be clearer if we just had a function:

```java
// Gets the portion of the requested bytes that can be written before the channel is
// no longer writable. 
int writableBytes(int requestedBytes) {
    return min(requestedBytes, chunkSize());
}
```

Then we just call this instead of `resetConnectionWindowHandicap`.